paymentsheet/src/main/java/com/stripe/android/paymentsheet/model/SupportedPaymentMethodKtx.kt

```diff
@@ -29,14 +30,45 @@ internal fun SupportedPaymentMethod.getPMAddForm(
 ) = requireNotNull(getSpecWithFullfilledRequirements(stripeIntent, config))
 
 /**
- * This function will determine if there is a valid for the payment method
+ * This function will determine if there is a valid spec for the payment method
  * given the [PaymentSheet.Configuration] and the [StripeIntent]
  */
 internal fun SupportedPaymentMethod.getSpecWithFullfilledRequirements(
     stripeIntent: StripeIntent,
     config: PaymentSheet.Configuration?
 ): LayoutFormDescriptor? {
-    val formSpec = formSpec
+    if (!stripeIntent.paymentMethodTypes.contains(code)) {
+        return null
+    }
+
+    val containsValidShippingInfo = when (stripeIntent) {
+        is PaymentIntent -> stripeIntent.containsValidShippingInfo
+        is SetupIntent -> false
+    }
```

Comment:
Now that we’re using `PaymentIntent` and `SetupIntent` everywhere, can we undo these changes again?